source/integrations/fastapi-integration.txt

```diff
@@ -0,0 +1,467 @@
+   * - ``PUT /students/{id}``
+     - Update a student
+
+   * - ``DELETE /students/{id}``
+     - Delete a student
+
+.. procedure::
+   :style: connected
+
+   .. step:: Student Routes
+
+    The ``create_student`` route receives the new student data as a JSON string
+    in a ``POST`` request. We must decode this JSON request body into a Python
+    dictionary before passing it to our MongoDB client. 
+
+    The ``insert_one`` method response includes the ``_id`` of the newly created
+    student (provided as ``id`` because this endpoint specifies
+    ``response_model_by_alias=False`` in the ``post`` decorator call. After we
+    insert the student into our collection, we use the ``inserted_id`` to find
+    the correct document and return this in our ``JSONResponse``.
+
+    FastAPI returns an HTTP ``200`` status code by default, but we will return a
+    ``201`` to explicitly that indicate the student has been created.
+    
+    Define the ``create_student`` route using the following code:
+
+    .. code-block:: python
+
+        @app.post(
+            "/students/",
+            response_description="Add new student",
+            response_model=StudentModel,
+            status_code=status.HTTP_201_CREATED,
+            response_model_by_alias=False,
+        )
+        async def create_student(student: StudentModel = Body(...)):
+            """
+            Insert a new student record. 
+        
+            A unique ``id`` will be created and provided in the response.
+            """
+            new_student = await student_collection.insert_one(
+                student.model_dump(by_alias=True, exclude=["id"])
+            )
+            created_student = await student_collection.find_one(
+                {"_id": new_student.inserted_id}
+            )
+            return created_student
+
+   .. step:: Read Routes
+
+    The application has two read routes: one for viewing all students, and one
+    for viewing an individual student specified by their ``id``.
+    
+    Define the ``list_students`` route to view all students using the following code:
+
+    .. code-block:: python
+
+        @app.get(
+            "/students/",
+            response_description="List all students",
+            response_model=StudentCollection,
+            response_model_by_alias=False,
+        )
+        async def list_students():
+            """
+            List all the student data in the database. 
+
+            The response is unpaginated and limited to 1000 results.
+            """
+            return StudentCollection(students=await student_collection.find().to_list())
+
+    .. note:: Results Pagination
+
+       This example uses the ``to_list()`` method; but in a real application, we
+       recommend using the `skip and limit parameters
+       <https://pymongo.readthedocs.io/en/stable/api/pymongo/asynchronous/collection.html#pymongo.asynchronous.collection.AsyncCollection.find>`__
+       in ``find`` to paginate your results. 
```

Comment:
S: not sure why formatting is off here - also use __ at the end of urls (2 underscores)